packages/strapi-design-system/.storybook/main.js

```diff
@@ -7,6 +7,7 @@ module.exports = {
     // Fix that allows to fix storybook issues with core-js
     // https://github.com/storybookjs/storybook/issues/11255
     config.resolve.alias['core-js'] = path.dirname(require.resolve('core-js'));
+    config.resolve.alias['@strapi/icons'] = path.dirname(require.resolve('../../strapi-icons/dist'));
```

Comment:
I don't think you need the `/dist` if the `main` key of the strapi-icons package is already the dist one.